src/rules/selector-pseudo-element-colon-notation/__tests__/index.js

```diff
@@ -0,0 +1,47 @@
+import { ruleTester } from "../../../testUtils"
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule("single", tr => {
+  tr.ok("")
+  tr.ok("a { color: pink; }")
+  tr.ok("::selection { color: pink; }")
+  tr.ok("a::spelling-error { color: pink; }")
+  tr.ok("a::grammar-error { color: pink; }")
+  tr.ok("li::marker { font-variant-numeric: tabular-nums; }")
+  tr.ok("input::placeholder { color: pink; }")
+
+  tr.ok("a:before { color: pink; }")
+  tr.ok("a:after { color: pink; }")
+  tr.ok("a:first-letter { color: pink; }")
+  tr.ok("a:first-line { color: pink; }")
+  tr.ok("a:before, a[data-before='before'] { color: pink; }")
+
+  tr.notOk("a::before { color: pink; }", messages.expected("single"))
+  tr.notOk("a::after { color: pink; }", messages.expected("single"))
+  tr.notOk("a::first-line { color: pink; }", messages.expected("single"))
+  tr.notOk("a::first-letter { color: pink; }", messages.expected("single"))
+})
+
+testRule("double", tr => {
+  tr.ok("")
+  tr.ok("a { color: pink; }")
+
+  tr.ok("::selection { color: pink; }")
```

Comment:
Here (and maybe in the "single" tests as well) we may want to throw in tests for pseudo-classes `:hover`, `:focus`, `:active`, because they are very common and should always use only one colon. Looks like you test that new pseudo-elements work only working with double-colons are accepted regardless of the expectation; so we should do the same with pseudo-classes that need to be accepted with single-colons no matter what.